lib/rubo_claus.rb

```diff
@@ -29,6 +29,11 @@ def catch_all(_proc)
     end
   end
 
+  private def execute_function(matching_function, runtime_args)
+    raise NoPatternMatchError unless matching_function
+    matching_function.call(*runtime_args)
+  end
+
```

Comment:
I also [touched this ternary a bit in my branch](https://github.com/mojotech/rubo_claus/blob/ae7300de5dabc647dcc7dcc07d2462797b45b1a6/lib/rubo_claus.rb#L18).  What do you think?  Passing the proc and args to another function call seems a bit overkill to me personally.